.gitignore

```diff
@@ -1 +1,2 @@
 result*
+/.pre-commit-config.yaml
```

Comment:
Is this file being added in this PR?

Reply:
This file gets generated once you enter the nix-shell, and it links to a file in the `nix-store`, so its best to git ignore it!

Reply:
Got it. Alright then!